File: rick.lua
local handle = io.popen("curl ascii.live/rick")
local result = handle:read("*a")
local success = handle:close()

if success then
    print(result)  -- Print command output
else
    print("Command failed")
    open_rickroll()
end

function open_rickroll()
    local url = "https://www.youtube.com/watch?v=dQw4w9WgXcQ"
    local os_type = os.getenv("OS")

    if os_type == "Windows_NT" then
        os.execute("start " .. url)  -- Windows
    elseif package.config:sub(1, 1) == '\\' then
        os.execute("open .. " .. url)  -- macOS
    else
        os.execute("xdg-open " .. url)  -- Assume Linux
    end
end

File: rick.rs
use std::process::{Command, exit};

fn main() {
    // Command to run
    let output = Command::new("curl")
        .arg("ascii.live/rick")
        .output();

    match output {
        Ok(output) => {
            if output.status.success() {
                // Print command output
                println!("{}", String::from_utf8_lossy(&output.stdout));
            } else {
                // Print error output if the command fails
                eprintln!("Command failed: {}", String::from_utf8_lossy(&output.stderr));
                open_rickroll();
            }
        }
        Err(e) => {
            eprintln!("Failed to execute command: {}", e);
            open_rickroll();
        }
    }
}

fn open_rickroll() {
    let url = "https://www.youtube.com/watch?v=dQw4w9WgXcQ";
    if cfg!(target_os = "windows") {
        // Windows
        Command::new("rundll32")
            .arg("url.dll,FileProtocolHandler")
            .arg(url)
            .spawn()
            .expect("Failed to open browser");
    } else if cfg!(target_os = "macos") {
        // macOS
        Command::new("open")
            .arg(url)
            .spawn()
            .expect("Failed to open browser");
    } else {
        // Assume Linux
        Command::new("xdg-open")
            .arg(url)
            .spawn()
            .expect("Failed to open browser");
    }
}